dateparser/languages/validation.py

```diff
@@ -0,0 +1,310 @@
+# -*- coding: utf-8 -*-
+import re
+from logging import getLogger
+
+
+class LanguageValidator(object):
```

Comment:
This class feels to me overly-complicated and with lots of duplication.
Do we really need all this?

This is like re-parsing all the file again and checking types and values.
If we really want to go this route, maybe a XML file with a schema would be a better format -- although personally, I don't see the point.

I get that we want to show an informative message if there is something wrong in the YAML file, but  maintaining this will be a real burden.

I propose keeping the validation code minimal and use assertions in the code to prevent wrong types and values.

Makes sense?